src/AzureExtension/DevBox/DevBoxProvider.cs

```diff
@@ -180,6 +180,10 @@ public IAsyncOperation<ComputeSystemsResult> GetComputeSystemsAsync(IDeveloperId
                 {
                     errorMessage = Resources.GetResource(Constants.NoDefaultUserFailKey);
                 }
+                else if (ex.Message.Contains("WAM Error"))

```

Comment:
Only comment here would be if you think we should explicitly try to check for both 
`MsalClientException`s  and `MsalServiceException`s  in a catch block? Just reading this https://learn.microsoft.com/en-us/dotnet/api/microsoft.identity.client.msalexception?view=msal-dotnet-latest 

I'm wondering if we should do something like

```C#
try
{
// ...
}
catch (MsalClientException msalClientEx)
{
 // handle MsalClientException  exception
}
catch (MsalServiceException msalServiceEx)
{
 // handle MsalServiceException exception
}
catch (Exception ex)
{
 // handle generic exception
}
```

Or is the issue that the exceptions we receive here aren't of those types so we have to attempt to parse the error message to confirm if it has to do with a WAM Error or not?

Reply:
Yeah, these are rethrown exceptions. So keeping the current change.